plugin/trino-mysql/src/main/java/io/trino/plugin/mysql/MySqlClientModule.java

```diff
@@ -73,6 +73,7 @@ public static Properties getConnectionProperties(MySqlConfig mySqlConfig)
         connectionProperties.setProperty("characterEncoding", "utf8");
         connectionProperties.setProperty("tinyInt1isBit", "false");
         connectionProperties.setProperty("rewriteBatchedStatements", "true");
+        connectionProperties.setProperty("connectionTimeZone", "SERVER");
```

Comment:
Also add `preserveInstants=true` - even though it's the default we want to be able to override a user manually setting it in their connection url since correctness depends on it.

Also a comment linking to "Solution 2a" at https://dev.mysql.com/doc/connector-j/8.0/en/connector-j-time-instants.html would be helpful here for reference.